draft-ietf-quic-qpack.md

```diff
@@ -219,10 +199,51 @@ In this way, even if request or push stream headers are decoded in a different
 order than encoded, the absolute indices will still identify the correct table
 entries.
 
-It is an error if the HPACK decoder encounters an indexed representation that
+It is an error if the decoder encounters an indexed representation that
 refers to an entry missing from the table, and the connection MUST be closed
 with the `HTTP_HPACK_DECOMPRESSION_FAILED` error code.
 
+## Single pass encoding
+
+An encoder making a single pass over a list of headers must choose Base Index
+before knowing Largest Reference.  An exception occurs when trying to reference
+a header inserted to the table after encoding has begun.  In this case,
+`relativeIndex = baseIndex - entry.absoluteIndex + staticTable.size` would
+either appear to be a static table reference, or a negative number that cannot
+be encoded using HPACK integer representation.  Instead, the index is encoded as
+`aboveBaseIndex = entry.absoluteIndex - baseIndex + staticTable.size`, and is
+used with different instructions that tell the decoder to use a different
+mechanism to calculate an entry's absolute index.
+
+### Literal with Above Base Name Index
+
+Literal with Above Base Name Index reuses the Literal with Incremental
+Indexing instruction, and is represented as follows:
+
+~~~~~~~~~~  drawing
+    0 1 2 3 4 5 6 7
+   +-+-+-+-+-+-+-+-+
+   |0|1| Index (6+)|
+   +---------------+
+   |H|Value Len(8+)|
+   +---------------+
+   |Value (*)      |
+   +---------------+
+~~~~~~~~~~
+
+### Indexed Header Field with Above Base Index
+
+Indexed Header Field with Above Base Index reuses the Table Size Update
+instruction, and is represented as follows:
+
+~~~~~~~~~~  drawing
+    0 1 2 3 4 5 6 7
+   +-+-+-+-+-+-+-+-+
+   |0|0|1|Index(5+)|
+   +---------------+
+~~~~~~~~~~
+
+
```

Comment:
I feel like these need more description in the instruction, even though you have a generally good description up above.  Maybe just some relocation/duplication of text?